modules/reactor/doc/content/source/meshgenerators/PatternedCartesianMeshGenerator.md

```diff
@@ -0,0 +1,50 @@
+
+## Geometry Information
+
+In order to generate the square patterned mesh, three fundamental parameters are needed:
+
+- [!param](/Mesh/PatternedCartesianMeshGenerator/inputs): a vector of mesh generator names that will be used as elements to form the patterned mesh;
+- [!param](/Mesh/PatternedCartesianMeshGenerator/pattern): a two-dimensional vector that represents the pattern of mesh to be generated. The elements must be integers from 0 to N-1, where N is the length of vector parameter [!param](/Mesh/PatternedCartesianMeshGenerator/inputs).
+- [!param](/Mesh/PatternedCartesianMeshGenerator/pattern_boundary): a MooseEnum parameter that can be either `square` or `none`. When `none` is selected, this object only stitches [!param](/Mesh/PatternedCartesianMeshGenerator/inputs) meshes into a patterned mesh without extraneous geometry, resulting in a simply stacked square of unit squares. When `square` is selected, a background region is added so that the generated mesh has an expanded sqaure shape instead of a simply stacked one. Concentric duct regions may also be optionally added to the sqaure periphery when this parameter is `square`. The user can refer to [Figure 1](#pattern_cart) for more details.
+
+!media reactor/meshgenerators/pattern_cart.png
+      style=display: block;margin-left:auto;margin-right:auto;width:40%;
+      id=pattern_cart
+      caption=A schematic showing the difference between `none` and `expanded` `pattern_boundary`.
+
+When [!param](/Mesh/PatternedCartesianMeshGenerator/pattern_boundary) is set as `square`, the user can also provide [!param](/Mesh/PatternedCartesianMeshGenerator/square_size) in either `apothem` or `radius` style to add external duct regions to the generated square mesh. Also, [!param](/Mesh/PatternedCartesianMeshGenerator/square_size) must also be provided to define the external square boundary size, which can be either `apothem` or `radius` of the square, as determined by [!param](/Mesh/PatternedCartesianMeshGenerator/square_size_style). In some cases, if [!param](/Mesh/PatternedCartesianMeshGenerator/square_size) is small enough, the square external boundary may cut off some of the stitched square meshes. As long as the concentric circular (`rings`) regions are not cut off, the rest of the mesh is deformed to accommodate the limited external boundaries. Users can also set [!param](/Mesh/PatternedCartesianMeshGenerator/deform_non_circular_region) as `false` to prevent the non-circular regions from being deformed.
```

Comment:
apothem? for cartesian?

Reply:
I will use simply `square_size` for the assembly dimension. But I think apothem/radius is still valid for `duct_sizes`